db/sovereign-db/src/schema/tables.rs

```diff
@@ -109,31 +109,31 @@ macro_rules! define_table_without_codec {
 /// )
 /// ```
 macro_rules! define_table_with_default_codec {
-	($(#[$docs:meta])+ ($table_name:ident) $key:ty => $value:ty) => {
-		define_table_without_codec!($(#[$docs])+ ( $table_name ) $key => $value);
+    ($(#[$docs:meta])+ ($table_name:ident) $key:ty => $value:ty) => {
+        define_table_without_codec!($(#[$docs])+ ( $table_name ) $key => $value);
 
-		impl ::sovereign_sdk::db::KeyEncoder<$table_name> for $key {
-			fn encode_key(&self) -> ::std::result::Result<::sovereign_sdk::maybestd::vec::Vec<u8>, ::sovereign_sdk::db::errors::CodecError> {
-				::std::result::Result::Ok(<Self as ::sovereign_sdk::serial::Encode>::encode_to_vec(self))
-			}
-		}
+        impl ::sovereign_sdk::db::KeyEncoder<$table_name> for $key {
+            fn encode_key(&self) -> ::std::result::Result<::sovereign_sdk::maybestd::vec::Vec<u8>, ::sovereign_sdk::db::errors::CodecError> {
+                <Self as ::borsh::BorshSerialize>::try_to_vec(self).map_err(|e| e.into())
```

Comment:
We can do `::std::result::Result::Ok(<Self as ::borsh::BorshSerialize>::try_to_vec(self)?)` instead of `map_err`

Reply:
But `try_to_vec` returns `std::io::Err` and we want CodecError. Does `?` do type conversion?